batch2/batch/front_end/front_end.py

```diff
@@ -287,10 +303,16 @@
                                              jobs_args)
                 async with conn.cursor() as cursor:
                     await cursor.executemany('''
-INSERT INTO `jobs-parents` (batch_id, job_id, parent_id)
+INSERT INTO `job_parents` (batch_id, job_id, parent_id)
 VALUES (%s, %s, %s);
 ''',
-                                             jobs_parents_args)
+                                             job_parents_args)
+                async with conn.cursor() as cursor:
+                    await cursor.executemany('''
+INSERT INTO `job_attributes` (batch_id, job_id, key, value)
```

Comment:
Sorry. I just thought of this. A future improvement might be to have all of this as one transaction in case something fails partway through. Future PR?

Reply:
I don't understand, it is one transaction.  I'm doing `conn.begin()` above and `conn.commit()` below.

Reply:
Sorry. Didn't see that. I assumed it was autocommit=True.

Reply:
It is true, but when you explicitly start a transaction with autocommit true, you get a transaction.

Reply:
ok. I think I understand now. It's because you have `.begin()` that you get a transaction and you explicitly have to commit it to end the transaction even though autocommit=True. I couldn't find documentation that was clear on this.